src/main/java/au/org/ala/biocache/dto/SearchRequestParams.java

```diff
@@ -61,7 +61,7 @@
     /**  The query context to be used for the search.  This will be used to generate extra query filters based on the search technology */
     protected String qc = "";
     /** To disable facets */
-    protected Boolean facet = FacetThemes.facetDefault;
+    protected Boolean facet = null;
```

Comment:
facet = null breaks some webservices.

Instead of a ```FacetService``` and ```createSpatialSearchRequestParams()```, it may be better to put a ```@PostConstruct public void init()``` in ``OccurrenceController``, duplicating ```groupFacets()``` so as to init ```FacetThemes```. This new ```init()``` should do so with a ```new Thread```  so as to not hold up the start up of biocache-service when SOLR initially unavailable, e.g. SOLR is on the same tomcat instance as biocache-service.